pkg/controller/master/setting/http_proxy.go

```diff
@@ -18,7 +18,13 @@ const (
 func (h *Handler) syncHTTPProxy(setting *harvesterv1.Setting) error {
 	// Add envs to the backup secret used by Longhorn backups
 	var httpProxyConfig util.HTTPProxyConfig
-	if err := json.Unmarshal([]byte(setting.Value), &httpProxyConfig); err != nil {
+	// Convert an empty `Setting.Value` string to its JSON quasi-equivalent,
+	// an empty object, otherwise the decoding will fail.
+	var value = setting.Value
+	if value == "" {
+		value = "{}"
```

Comment:
Or should it be like that?
```suggestion
		value = setting.Default
```

Reply:
```
// general setting processing
// take the target value from `Value` or `Default` field, depends on if `Value` field is empty
value := setting.Value
if value == "" {
	value = setting.Default
}

// for this setting, empty is equal to and further converted to "{}", avoid JSON Unmarshal failure
if value == "" {
	value = "{}"
}

// ....
```